providers/common/kdfs/tls1_prf.c

```diff
@@ -137,38 +138,73 @@ static int kdf_tls1_prf_derive(void *vctx, unsigned char *key,
         ERR_raise(ERR_LIB_PROV, PROV_R_MISSING_SEED);
         return 0;
     }
-    return tls1_prf_alg(md, ossl_prov_digest_md(&ctx->sha1),
+
+    return tls1_prf_alg(ctx->P_hash, ctx->P_sha1,
                         ctx->sec, ctx->seclen,
                         ctx->seed, ctx->seedlen,
                         key, keylen);
 }
 
+static EVP_MAC_CTX *kdf_tls1_prf_mkmac(OPENSSL_CTX *libctx,
```

Comment:
Nit: The mkmac name seems slightly confusing to me as it creates MAC_CTX and not the actual mac. So I'd prefer something like kdf_tls1_prf_macctx or kdf_tls1_prf_mkmacctx. 
But this is just for your consideration the approval holds even without the change.

Reply:
I agree with you, so I'll rename